src/test/perl/lock.t

```diff
@@ -13,38 +13,10 @@ unlink(LOCK_TEST);
 my $lock=CAF::Lock->new(LOCK_TEST);
 
 ok(!$lock->is_locked(), "Unlocked at start");
-my $lockpid=$lock->get_lock_pid();
-
-is($lockpid, undef, "Lock PID undefined on unaquired lock");
 
 ok($lock->set_lock(), "Lock set");
 ok($lock->is_locked(), "Locked on request");
 
-is($lock->get_lock_pid(), $$, "Lock PID correctly set on locked object");
-
-ok(!$lock->is_stale(), "Lock is NOT stale");
-
 ok($lock->unlock(), "Lock released");
```

Comment:
take the lock with the seconds lock instances, it should work now (and that is the actual test that the lock was released)